api/core/app/entities/app_invoke_entities.py

```diff
@@ -116,13 +117,36 @@ class EasyUIBasedAppGenerateEntity(AppGenerateEntity):
     model_config = ConfigDict(protected_namespaces=())
 
 
-class ChatAppGenerateEntity(EasyUIBasedAppGenerateEntity):
+class ConversationAppGenerateEntity(AppGenerateEntity):
     """
-    Chat Application Generate Entity.
+    Base entity for conversation-based app generation.
     """
 
     conversation_id: Optional[str] = None
-    parent_message_id: Optional[str] = None
+    parent_message_id: Optional[str] = Field(
+        default=None,
+        description=(
+            "Starting from v0.9.0, parent_message_id is used to support message regeneration "
+            "for internal chat API. For service API, we need to ensure its forward compatibility, "
+            "so passing in the parent_message_id as request arg is not supported for now. "
+            "It will be set to UUID_NIL so that the subsequent processing will treat it as legacy messages."
+        ),
+    )
+
+    @field_validator("parent_message_id")
+    @classmethod
+    def validate_parent_message_id(cls, v, info: ValidationInfo):
+        if info.data.get("invoke_from") == InvokeFrom.SERVICE_API:
+            return UUID_NIL
```

Comment:
Modifying the input value within the validator is not an ideal approach. I recommend throwing an exception when v != UUID_NIL, and the user should ensure the accuracy of the input data.